### DISCUSSION TITLE
Master–detail interfaces

### DISCUSSION THREAD

**Discussion by crudyrudy** [Budibase/budibase#2462] [Ideas]
One of the most important forms of representation in my (planned) applications are the following:

1. under or next to a list of records, other fields of the same table (details) of the record currently selected in the list are displayed.

2. records from another table related to the currently selected record in the list are displayed under or next to the first list. For example, think of a list of companies and a list of employees working for the active company.

Are there already plans to offer such displays?

**Comment by oswaldB** [comment]
or also give us the opportunity to enlarge the left side bar :)
![image](https://user-images.githubusercontent.com/3023526/130791579-a3461470-0d6b-48f5-b45c-853ad19c1b42.png)

**Reply by joebudi** [reply]
Hey @oswaldB 

There is a PR for this: #2443

**Comment by joebudi** [comment]
@crudyrudy Both of your scenarios are currently possible but not trivial. We have plans to improve the UX around both scenarios.

For scenario one:
- create a new table and autogenerate screens 
- go to design and delete the screen which reads /table/:id - we don't need this as our list screen will now also be our detail screen
- Change the URL of your list screen /table to  /table/:id? 
- Add a new data provider to the screen above - point it to your table. 
- Click the filter button (a drawer will pop up), Select ID - Equals - Binding - {{ URL.id }}
![Screenshot 2021-08-25 at 14 35 51](https://user-images.githubusercontent.com/49767913/130800422-2ef4e34f-34b0-4c19-93ac-486eb67abc58.png)
- Add a repeater - point to data provider
- Add a form - Change type to Update (so we can also update the date) and change the schema to your table
- Add a field group and click update form fields (new fields will be generated from your table and added to your nav :-) )
- Add a new button, and then click the Define actions button in the props panel
- Click the Add actions button, and click Save row
- Select New Form, then select your table and click save
![Screenshot 2021-08-25 at 14 37 53](https://user-images.githubusercontent.com/49767913/130800757-66f4f22b-9d8e-4392-8964-9772cc263a86.png)

Preview your app and that should be it.

For scenario two:
- if you autogenerated screens from a table in your data section you will have a detail screen - /table/:id.
- Within this screen, you will need to add an additional data provider 
- In the property's panel, where it says Data, select (under relationship) your relationship Companies.Employees.

The data / tree should follow this:
`Data Provider (Companies) > Repeater > Data Provider (Companies.Employees)
`


I hope this helps!

**Comment by crudyrudy** [comment]
Thank you for this detailed answer. I've already tried the solution for scenario two - works great!
